app/listeners/object_event_listener.rb

```diff
@@ -0,0 +1,160 @@
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.campaign_gift_purchase_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.campaign_gift_purchase_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.campaign_gift_purchase_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.event_discount_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.event_discount_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.event_discount_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_level_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_level_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_level_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_purchase_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_purchase_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.ticket_purchase_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_address_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_address_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_address_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_note_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_note_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.supporter_note_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.donation_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.donation_updated(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.donation_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.custom_field_definition_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.custom_field_definition_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.tag_definition_created(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  def self.tag_definition_deleted(event)
+    enqueue_transmissions_to_webhooks(event)
+  end
+
+  private
+
+  def self.enqueue_transmissions_to_webhooks(event)
+    object_event_hook_configs(event).each do |config|
+      WebhookTransmitJob.perform_now(config.webhook, event)
```

Comment:
Two things: 

* You should enqueue each of these to their own job. Right now, if one of the transmissions fails, then other transmissions fail for that event.
*You should pass the ObjectEventHookConfig to the job itself. ActiveRecord records can be serialized properly for ActiveJobs but regular objects from classes can't.

Reply:
What do you mean by their own job? I thought that by making one `perform_now` for each config it would work.

Reply:
`.perform_now` is the equivalent of saying "run this code right now on the current thread and block until it finishes". `.perform_later` means "queue an ActiveJob onto the proper queue adapter and then don't wait for it to finish". `.perform_now` is synchronous and `perform_later` is async.

Reply:
Thanks!